CONTRIBUTING.md

```diff
@@ -0,0 +1,33 @@
+# `Helix-editor`
+### First off, Thanks for contributing! Please read the following: 
+
+Contributors are very welcome! **No contribution is too small and all contributions are valued.**
+
+Some suggestions to get started:
+
+- You can look at the [good first issue](https://github.com/helix-editor/helix/labels/E-easy) label on the issue tracker.
+- Help with packaging on various distributions needed!
+- To use print debugging to the `~/.cache/helix/helix.log` file, you must:
+  * Print using `log::info!`, `warn!`, or `error!`. (`log::info!("helix!")`)
+  * Pass the appropriate verbosity level option for the desired log level. (`hx -v <file>` for info, more `v`s for higher severity inclusive)
+- If your preferred language is missing, integrating a tree-sitter grammar for
+    it and defining syntax highlight queries for it is straight forward and
+    doesn't require much knowledge of the internals.
+
+We provide an [architecture.md](./docs/architecture.md) that should give you
+a good overview of the internals.
+
+## Before contributing:
+- Please check to see all closed and open issues and/or discussions that what problem you are facing has already been discussed, or solved. Please make sure you have the latest version.
+- If you have a question or a doubt. Please *DO NOT* open an issue. The [discussions](https://github.com/helix-editor/helix/discussions) tab is exactly for that.
```

Comment:
Some nit-picks about the phrasing.  I think this reads a little better:

> - Before submitting an issue, please search both closed and open issues to check that a similar issue hasn't already been submitted.
> - If you have questions about Helix's behavior or usage, please do not file an issue.  Instead, ask your questions in [discussions](https://github.com/helix-editor/helix/discussions).

For the second item, I also wonder if we should direct people to Helix's Matrix channel.

Reply:
Yeah.. That's better. Should I still mention: 
```
Please make sure you have the latest version.
```
?

I still haven't edited it. After you say, I'll do the same.

Reply:
Oh, yeah.  For filing issues that's probably good to mention.  Good catch!  So maybe:

> - Before submitting an issue, please make sure you're using the latest version of Helix, and search both closed and open issues to check that a similar issue hasn't already been submitted.

Reply:
Done the same 👍 Added a blockquote at the top of the README. please have a look(again) ;-)